### PR TITLE
Fix mod_random.py that is broken for CentOS 6

### PR DESCRIPTION
**Pull request by danlsgiga**
mod_random.py is currently broken in my CentOS 6 boxes when running Salt 2015.8.0. Doing the proposed change fix the issue. Finding a better way to validate this function on Debian systems would be great!

### CODE REVIEW

**Comment by jtand** [comment]
Go Go Jenkins!

**Comment by cachedout** [comment]
@danlsgiga Could you provide a little more context on what's breaking on CentOS6 and why this fixes it? I'd like to understand a little more about what's happening on your system before we merge this.

**Comment by danlsgiga** [comment]
Hey @cachedout, sure!

So, I was trying to use the mod_random last week specifically to use the seed function to have an unique ID for each server in my cluster. I'm working with CentOS 7 and CentOS 6 for testing purposes since those are the OS / Versions we currently support in our company. When my tests failed only in the CentOS 6 boxes for the use of any function inside the mod_random module I verified that the 'algorithms' dictionary doesn't exist in CentOS 6 while it works fine in CentOS 7.

From my understanding this dictionary is present only when using Python 2.7+, however CentOS 6 is officially currently using Python 2.6.6 while CentOS 7 has 2.7.5.

Running the following snippet in both OS gives me the following returns:

    from pprint import pprint
    import hashlib
    pprint(vars(hashlib))

CentOS7:

    ...
    '__file__': '/usr/lib64/python2.7/hashlib.pyc',
    '__name__': 'hashlib',
    '__package__': None,
    '_hashlib': <module '_hashlib' from '/usr/lib64/python2.7/lib-dynload/_hashlib.so'>,
    'algorithms': ('md5', 'sha1', 'sha224', 'sha256', 'sha384', 'sha512'),
    'md5': <built-in function openssl_md5>,
    'new': <function __hash_new at 0x7f00743efe60>,
    'sha1': <built-in function openssl_sha1>,
    'sha224': <built-in function openssl_sha224>,
    'sha256': <built-in function openssl_sha256>,
    'sha384': <built-in function openssl_sha384>,
    'sha512': <built-in function openssl_sha512>
    ...

CentOS6:

    ...
    '__file__': '/usr/lib64/python2.6/hashlib.pyc',
    '__hash_new': <function __hash_new at 0x7f8fbf318320>,
    '__name__': 'hashlib',
    '__package__': None,
    '_hashlib': <module '_hashlib' from '/usr/lib64/python2.6/lib-dynload/_hashlib.so'>,
    'md5': <built-in function openssl_md5>,
    'new': <function __hash_new at 0x7f8fbf318320>,
    'sha1': <built-in function openssl_sha1>,
    'sha224': <built-in function openssl_sha224>,
    'sha256': <built-in function openssl_sha256>,
    'sha384': <built-in function openssl_sha384>,
    'sha512': <built-in function openssl_sha512>
    ...

So, basically, when running any function from mod_random in a CentOS 6 box, it will return False. This issue was introduced in Salt 2105.8.0 release.

Does that help you?

**Comment by cachedout** [comment]
Hi @danlsgiga .

Thanks for the explanation here. Much appreciated.

However, I'm not sure that this is a PR that we can merge. The `__virtual__` function is in place to determine whether or not the appropriate pre-requisites are available on a given system for the functions in the Salt module to operate properly.

As you've shown, the necessary functionality in `hashlib` isn't available on Python 2.6. 

If we merged this PR, the Salt module would certainly load but any use of `hashlib.algorithms` in the module would then stacktrace.

In my view, the `__virtual__` function is currently written correctly since it is doing its job of refusing to load the Salt module if the required libraries are not present.

Thoughts?

**Comment by danlsgiga** [comment]
Hi @cachedout... thanks for your quick feedbacks!
Well, that makes total sense. Since I was not using the hash function I didn't notice the stacktrace. Please disregard this PR. But we still need to fix this in some other way.
Do you see any options to do that without breaking compatibility?

**Comment by cachedout** [comment]
Hi @danlsgiga What we'd need to do here would be to investigate whether or not the module itself could be made to function using only code present in py2.6. That may or may not be possible and would likely involve finding a replacement for the missing functionality in hashlib for py2.6 installations.
